### PR TITLE
Change azure-sdk-for-go version to 0.4.0 and go version to be 1.18

### LINKED ISSUE #2886: Upgrade azure-sdk-for-go library to 4.0
### Describe your problem(s)
$subject needs to be done in order to fix the ASB client hanging issue.
https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azservicebus/CHANGELOG.md

### How will you implement it

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Issues

Fixes #2886 

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [ ] Assigned the project
- [x] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Review by AmaliMatharaarachchi — APPROVED** [review]

File: adapter/go.mod [hunk 1]
```diff
@@ -1,11 +1,10 @@
 module github.com/wso2/product-microgateway/adapter
 
-go 1.15
+go 1.18
```
Comment (AmaliMatharaarachchi) [inline adapter/go.mod:3]: shall we update develper guide/prerequisites as well

**Review by pubudu538 — APPROVED** [review]

File: adapter/go.mod (hunk 1 above)
Comment (VirajSalaka) [reply adapter/go.mod:3]: updated. Thanks

**Review by VajiraPrabuddhaka — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE2 cluster : dev-deployment-v2 : 20220506.9

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20220506.9

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE1 cluster : dev-deployment-v2 : 20220506.9

**Comment by choreo-cicd** [comment]
[failed] DataplaneE1 cluster : stage-deployment-v2 : 20220518.3

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE1 cluster : stage-deployment-v2 : 20220518.5

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20220523.1

**Pull request merged**